### ISSUE TITLE
Screen reader behaviour for labels in promo component

### ISSUE THREAD

**Issue by JoshTumath** [bbc/gel#131]
While implementing the Promo, I don't think the GEL documentation described the recommended markup for the Label, other than hiding it from screen readers.

I'd like to propose the experience would be something like this:

```html
<div class="gel-promo">
  <div class="gel-promo__content">
    <a class="gel-promo__headline" href="path/to/content">
      <strong class="gel-promo__label">New</strong>
      University Challenge
    </a>
  </div>
</div>
```

```css
.gel-promo__label {
  position: absolute;
  top: 0;
  left: 0;
}
```

The screen reader would read out the label within the headline, but the label would visually render in the top left/right corner of the promo.

**Comment by micmath** [comment]
This follows the advice in the GEL design guides:

>A label is a temporary indicator used to attract attention to a piece of content. It’s letting you know why you are seeing this. Typical uses are:
>State: To show where a piece of content is within the contents lifecycle, e.g. if a piece of content is currently in progress or due to expire. This is in place of or with a progress bar.
> Editorial: To show that there’s something special about that content i.e. ‘new’ or ‘exclusive’.

Would the CSS need to include background color, padding etc for the label?

**Comment by JoshTumath** [comment]
> Would the CSS need to include background color, padding etc for the label?

Yes, you'd still presumably add all of your visual styling. I just chose to keep my example code short.

However, I've been implementing this today and, in practice, I've found that I needed to do (the equivalent of) this instead:

```html
<div class="gel-promo">
  <div class="gel-promo__image">
    <strong class="gel-promo__label" aria-hidden="true">New</strong>
    <img src="path/to/image" alt="">
  </div>
  <div class="gel-promo__content">
    <a class="gel-promo__headline" href="path/to/content">
      <span class="gel-sr">New</span>
      University Challenge
    </a>
  </div>
</div>
```

This is because we want to prevent the Label from overflowing the `div.gel-promo__image`.

**Comment by micmath** [comment]
Would you be willing to modify the reference implementation in the documentation to include an example of this please? Pull requests are welcomed :)

**Comment by simonsinclair** [comment]
> However, I've been implementing this today and, in practice, I've found that I needed to do (the equivalent of) this instead:
> 
> ```
> <div class="gel-promo">
>   <div class="gel-promo__image">
>     <strong class="gel-promo__label" aria-hidden="true">New</strong>
>     <img src="path/to/image" alt="">
>   </div>
>   <div class="gel-promo__content">
>     <a class="gel-promo__headline" href="path/to/content">
>       <span class="gel-sr">New</span>
>       University Challenge
>     </a>
>   </div>
> </div>
> ```

Hey @JoshTumath, please can you help me understand how this is different from the first promo in [the reference implementation](https://bbc.github.io/gel/components/demos/promos/)?

**Comment by JoshTumath** [comment]
> Hey @JoshTumath, please can you help me understand how this is different from the first promo in [the reference implementation](https://bbc.github.io/gel/components/demos/promos/)?

Thanks for showing me that. I actually didn't notice that in the reference implementation. I was just looking at the documentation, which I don't think mentioned anything about it.

**Comment by simonsinclair** [comment]
No problem. As @micmath mentioned, we'd welcome a pull request if you'd like to see it mentioned in the documentation - the relevant file is [components/promos.md](https://github.com/bbc/gel/blob/master/src/components/promos.md). I think it'd be a good addition, because there's definitely some things to consider when implementing this little element.
